hikari/internal/collections.py

```diff
@@ -366,10 +344,7 @@ def get_index_or_slice(
     if isinstance(index_or_slice, slice):
         return tuple(itertools.islice(mapping.values(), index_or_slice.start, index_or_slice.stop, index_or_slice.step))
 
-    if isinstance(index_or_slice, int):
-        try:
-            return next(itertools.islice(mapping.values(), index_or_slice, None))
-        except StopIteration:
-            raise IndexError(index_or_slice) from None
-
-    raise TypeError(f"sequence indices must be integers or slices, not {type(index_or_slice).__name__}")
+    try:
+        return next(itertools.islice(mapping.values(), index_or_slice, None))
+    except StopIteration:
+        raise IndexError(index_or_slice) from None
```

Comment:
Docstring needs updating (forgot to do so)

Reply:
To what? TypeError can still be raised can't it?

Reply:
Yes, but not by us. We only document what we explicitly raise

Reply:
sussy